File: ios/Samples/Showcase-objc/LiferayScreens-Showcase-Objc/ViewControllers/AssetListScreenletViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
#import "AssetListScreenletViewController.h"
#import "AssetListScreenletInterop.h"

@interface AssetListScreenletViewController ()

@property (nonatomic, weak) IBOutlet AssetListScreenlet* screenlet;
@property (nonatomic, weak) IBOutlet UIPickerView* picker;

@property (nonatomic, assign) NSInteger selectedAsset;

@property (nonatomic, strong) NSArray *pickerData;

@end


@implementation AssetListScreenletViewController

- (IBAction)changeAssetAction:(id)sender {
	[self showPicker:NO animated:YES];
	self.screenlet.classNameId = self.selectedAsset;
	[self.screenlet loadList];
}

- (IBAction)openAssetPickerAction:(id)sender {
	[self showPicker:YES animated:YES];
}

- (IBAction)refreshControlChanged:(UISwitch *)sender {
	self.screenlet.refreshControl = sender.on;
}

- (void)viewDidLoad {
	[super viewDidLoad];

	self.screenlet.delegate = self;

	NSMutableArray *data = [[NSMutableArray alloc] init];

	[data addObject:@[@"Group", @(AssetClassNameIdGroup)]];
	[data addObject:@[@"Layout", @(AssetClassNameIdLayout)]];
	[data addObject:@[@"Organization", @(AssetClassNameIdOrganization)]];
	[data addObject:@[@"User", @(AssetClassNameIdUser)]];
	[data addObject:@[@"UserGroup", @(AssetClassNameIdUserGroup)]];
	[data addObject:@[@"BlogsEntry", @(AssetClassNameIdBlogsEntry)]];
	[data addObject:@[@"BookmarksEntry", @(AssetClassNameIdBookmarksEntry)]];
	[data addObject:@[@"BookmarksFolder", @(AssetClassNameIdBookmarksFolder)]];
	[data addObject:@[@"CalendarEvent", @(AssetClassNameIdCalendarEvent)]];
	[data addObject:@[@"DLFileEntry", @(AssetClassNameIdDLFileEntry)]];
	[data addObject:@[@"DLFileEntryMetadata", @(AssetClassNameIdDLFileEntryMetadata)]];
	[data addObject:@[@"DLFileEntryType", @(AssetClassNameIdDLFileEntryType)]];
	[data addObject:@[@"DLFileRank", @(AssetClassNameIdDLFileRank)]];
	[data addObject:@[@"DLFileShortcut", @(AssetClassNameIdDLFileShortcut)]];
	[data addObject:@[@"DLFileVersion", @(AssetClassNameIdDLFileVersion)]];
	[data addObject:@[@"DDLRecord", @(AssetClassNameIdDDLRecord)]];
	[data addObject:@[@"DDLRecordSet", @(AssetClassNameIdDDLRecordSet)]];
	[data addObject:@[@"JournalArticle", @(AssetClassNameIdJournalArticle)]];
	[data addObject:@[@"JournalFolder", @(AssetClassNameIdJournalFolder)]];
	[data addObject:@[@"MBMessage", @(AssetClassNameIdMBMessage)]];
	[data addObject:@[@"MBThread", @(AssetClassNameIdMBThread)]];
	[data addObject:@[@"MBCategory", @(AssetClassNameIdMBCategory)]];
	[data addObject:@[@"MBDiscussion", @(AssetClassNameIdMBDiscussion)]];
	[data addObject:@[@"MBMailingList", @(AssetClassNameIdMBMailingList)]];
	[data addObject:@[@"WikiPage", @(AssetClassNameIdWikiPage)]];
	[data addObject:@[@"WikiPageResource", @(AssetClassNameIdWikiPageResource)]];
	[data addObject:@[@"WikiNode", @(AssetClassNameIdWikiNode)]];

	self.pickerData = [[NSArray alloc] initWithArray:data];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
	self.selectedAsset = [self.pickerData[row][1] intValue];
	NSLog(@"[ASSET TYPE] Selected -> %ld", (long)self.selectedAsset);
}

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
	return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
	return self.pickerData.count;
}

-(NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
	return self.pickerData[row][0];
}

- (void)showPicker:(BOOL)show animated:(BOOL)animated {
	[UIView animateWithDuration:animated ? 0.5 : 0 animations:^{
		self.picker.superview.frame = CGRectMake(
				self.picker.superview.frame.origin.x,
				show ? 0 : self.view.frame.size.height,
				self.picker.superview.frame.size.width,
				self.picker.superview.frame.size.height);
	}];
}

- (void)screenlet:(AssetListScreenlet *)screenlet
		onAssetListResponseEntries:(NSArray *)entries {
	NSLog(@"DELEGATE onAssetListResponse -> %@", entries);
}

- (void)screenlet:(AssetListScreenlet *)screenlet
		onAssetListError:(NSError *)error {
	NSLog(@"DELEGATE onAssetListError -> %@", error);
}

- (void)screenlet:(AssetListScreenlet *)screenlet
		onAssetSelectedEntry:(AssetListScreenletEntry *)entry {
	NSLog(@"DELEGATE onAssetSelected -> %@", entry);
}


@end

File: ios/Samples/Showcase-swift/LiferayScreens-Showcase-Swift/AssetListScreenletViewController.swift
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
import UIKit

class AssetListScreenletViewController: UIViewController, AssetListScreenletDelegate,
	UIPickerViewDelegate, UIPickerViewDataSource {

	@IBOutlet var screenlet: AssetListScreenlet?
	@IBOutlet var picker: UIPickerView?

	var selectedAsset = AssetClassNameId.Group

	var pickerData:[(name:String, assetId:AssetClassNameId)] = []

	@IBAction func changeAssetAction(sender: AnyObject) {
		showPicker(false, animated:true)
		screenlet?.classNameId = selectedAsset.rawValue
		screenlet?.loadList()
	}

	@IBAction func openAssetPickerAction(sender: AnyObject) {
		showPicker(true, animated:true)
	}

	@IBAction func refreshControlChanged(sender: UISwitch) {
		self.screenlet?.refreshControl = sender.on
	}

	func pickerView(pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
		selectedAsset = pickerData[row].assetId
		println("[ASSET TYPE] -> Selected \(selectedAsset.rawValue)")
	}

    func numberOfComponentsInPickerView(pickerView: UIPickerView) -> Int {
		return 1
	}

    func pickerView(pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
		return pickerData.count
	}

	func pickerView(pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String! {
		return pickerData[row].name
	}

	override func viewDidLoad() {
		super.viewDidLoad()

		pickerData.append(name:"Group", assetId:.Group)
		pickerData.append(name:"Layout", assetId:.Layout)
		pickerData.append(name:"Organization", assetId:.Organization)
		pickerData.append(name:"User", assetId:.User)
		pickerData.append(name:"UserGroup", assetId:.UserGroup)
		pickerData.append(name:"BlogsEntry", assetId:.BlogsEntry)
		pickerData.append(name:"BookmarksEntry", assetId:.BookmarksEntry)
		pickerData.append(name:"BookmarksFolder", assetId:.BookmarksFolder)
		pickerData.append(name:"CalendarEvent", assetId:.CalendarEvent)
		pickerData.append(name:"DLFileEntry", assetId:.DLFileEntry)
		pickerData.append(name:"DLFileEntryMetadata", assetId:.DLFileEntryMetadata)
		pickerData.append(name:"DLFileEntryType", assetId:.DLFileEntryType)
		pickerData.append(name:"DLFileRank", assetId:.DLFileRank)
		pickerData.append(name:"DLFileShortcut", assetId:.DLFileShortcut)
		pickerData.append(name:"DLFileVersion", assetId:.DLFileVersion)
		pickerData.append(name:"DDLRecord", assetId:.DDLRecord)
		pickerData.append(name:"DDLRecordSet", assetId:.DDLRecordSet)
		pickerData.append(name:"JournalArticle", assetId:.JournalArticle)
		pickerData.append(name:"JournalFolder", assetId:.JournalFolder)
		pickerData.append(name:"MBMessage", assetId:.MBMessage)
		pickerData.append(name:"MBThread", assetId:.MBThread)
		pickerData.append(name:"MBCategory", assetId:.MBCategory)
		pickerData.append(name:"MBDiscussion", assetId:.MBDiscussion)
		pickerData.append(name:"MBMailingList", assetId:.MBMailingList)
		pickerData.append(name:"WikiPage", assetId:.WikiPage)
		pickerData.append(name:"WikiPageResource", assetId:.WikiPageResource)
		pickerData.append(name:"WikiNode", assetId:.WikiNode)

		self.screenlet?.delegate = self
	}

	func onAssetListResponse(entries:[AssetListScreenletEntry]) {
		println("DELEGATE: onAssetListResponse called -> \(entries)");
	}

	func onAssetListError(error: NSError) {
		println("DELEGATE: onAssetListError called -> \(error)");
	}

	func onAssetSelected(entry: AssetListScreenletEntry) {
		println("DELEGATE: onAssetSelected called -> \(entry)");
	}

	func showPicker(show:Bool, animated:Bool) {
		UIView.animateWithDuration(animated ? 0.5 : 0) {
			self.picker!.superview!.frame = CGRectMake(
				self.picker!.superview!.frame.origin.x,
				show ? 0 : self.view.frame.size.height,
				self.picker!.superview!.frame.size.width,
				self.picker!.superview!.frame.size.height)
		}
	}

}
